microsoft-edge/progressive-web-apps-chromium/how-to/push.md

```diff
@@ -6,155 +6,141 @@ ms.author: msedgedevrel
-                    userVisibleOnly: true,
-                    applicationServerKey: urlBase64ToUint8Array(vapidPublicKey)
-                });
-            });
+async function subscribeToPushMessages() {
+  const serviceWorkerRegistration = await navigator.serviceWorker.ready;
+
+  // Check if the user has an existing subscription
+  let pushSubscription = serviceWorkerRegistration.pushManager.getSubscription();
+  if (pushSubscription) {
+    // The user is already subscribed to push notifications
+    return;
+  }
+
+  try {
+    // Subscribe the user to push notifications
+    pushSubscription = await serviceWorkerRegistration.pushManager.subscribe({
+      userVisibleOnly: true,
+      applicationServerKey: urlBase64ToUint8Array("YOUR PUBLIC VAPID KEY HERE")
     });
+  } catch (err) {
+    // The subscription wasn't successful.
+    console.log("Error", err);
+  }
+}
 
 // Utility function for browser interoperability
 function urlBase64ToUint8Array(base64String) {
-    var padding = '='.repeat((4 - base64String.length % 4) % 4);
-    var base64 = (base64String + padding)
-        .replace(/\-/g, '+')
-        .replace(/_/g, '/');
-
-    var rawData = window.atob(base64);
-    var outputArray = new Uint8Array(rawData.length);
-
-    for (var i = 0; i < rawData.length; ++i) {
-        outputArray[i] = rawData.charCodeAt(i);
-    }
-    return outputArray;
+  var padding = '='.repeat((4 - base64String.length % 4) % 4);
+  var base64 = (base64String + padding)
+      .replace(/\-/g, '+')
+      .replace(/_/g, '/');
+  
+  var rawData = window.atob(base64);
+  var outputArray = new Uint8Array(rawData.length);
+  
+  for (var i = 0; i < rawData.length; ++i) {
+    outputArray[i] = rawData.charCodeAt(i);
+  }
+  return outputArray;
 }
 ```
 
-See also [PushManager](https://developer.mozilla.org/docs/Web/API/PushManager) and [Web-Push](https://www.npmjs.com/package/web-push#usage).
+The VAPID key mentioned in the previous code snippet is a public key that is used to identify the server that sends the push messages and encrypt the push message payload.  See [Step 3 - Send push messages from your server](#step-3---send-push-messages-from-your-server) for more information about VAPID keys.
 
 
 <!-- ====================================================================== -->
-## Step 3 - Listen for push notifications
+## Step 3 - Send push messages from your server
 
-After a subscription is created in your PWA, add handlers to the service worker to respond to push events.  Push event are sent from the server to display toast notifications.  Toast notifications display data for a received message.  To do any of the following tasks, you must add a `click` handler:
+Your application needs VAPID (Voluntary Application Server Identification) keys in order to send push messages from your server to your PWA clients.  There are several VAPID key generators available online (for example, [vapidkeys.com](https://vapidkeys.com)).
```

Comment:
```suggestion
Your application needs VAPID (Voluntary Application Server Identification) keys in order to send push messages from your server to your PWA clients.  There are several VAPID key generators available online (for example, [Vapidkeys.com](https://vapidkeys.com)).
```
https://learn.microsoft.com/en-us/style-guide/urls-web-addresses#references-to-specific-sites-and-domains - "To refer to an entire website or top-level domain, such as Microsoft.com, omit http://www from the URL and capitalize only the first letter of the URL, even if the site name is capitalized differently."